CMakeLists.txt

```diff
@@ -1,6 +1,9 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(network_interface)
 
+add_definitions(-std=c++11 -Wall -Wextra -DBUILD_ROS)
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
```

Comment:
Why add pthread here? The <thread> header / C++ std::thread should take care of this.